safekeeper/src/receive_wal.rs

```diff
@@ -498,7 +535,7 @@ impl WalAcceptor {
         // we will send keepalives by replying to these requests once per second.
         let mut next_keepalive = Instant::now();
 
-        loop {
+        while !self.tli.tli.cancel.is_cancelled() {
```

Comment:
Same as above: this will block on inner awaits, consider a `select`.

Reply:
I'm refactoring `WalAcceptor` to be event-driven over in #9462, so I can slot in the cancellation there.

Reply:
I merged main into this branch, and in the process added the cancellation case to everywhere we use select! in this file